support-frontend/assets/components/paymentButton/defaultPaymentButtonContainer.tsx

```diff
@@ -62,9 +63,19 @@ export function DefaultPaymentButtonContainer({
 			  )
 			: getUserSelectedAmount(state),
 	);
+	const promotion = isSupporterPlus
+		? useContributionsSelector((state) =>
+				getPromotion(
+					state.page.checkoutForm.product.productPrices,
+					countryId,
+					state.page.checkoutForm.product.billingPeriod,
+				),
+		  )
+		: undefined;
+	const amount = promotion?.discountedPrice ?? selectedAmount;
```

Comment:
The pay button is responsible for working out it's total to display, so we need to do this again here.

Reply:
understand, not sure why this was not wrong in the past however? Maybe one o fthe previous PR's brought this on?